shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -184,6 +184,152 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
     case _ => throw CodeGenError(s"ill-formed application $term")
   }
 
+  private def createASTCall(tp: Str, args: Ls[Term]): App =
+    App(Var(tp), Tup(args.map(a => N -> Fld(FldFlags.empty, a))))
+
+  class FreeVars(val vs: Set[Str])
+
+  // * Left: the branch is quoted and it has been desugared
+  // * Right: the branch is not quoted and quoted subterms have been desugared
+  private def desugarQuotedBranch(branch: CaseBranches)(
+    implicit scope: Scope, isQuoted: Bool, freeVars: FreeVars
+  ): Either[Term, CaseBranches] = branch match {
+    case Case(pat, body, rest) =>
+      val dp = desugarQuote(pat)
+      val db = desugarQuote(body)
+      desugarQuotedBranch(rest) match {
+        case L(t) => L(createASTCall("Case", dp :: db :: t :: Nil))
+        case R(b) => dp match {
+          case dp: SimpleTerm => R(Case(dp, db, b))
+          case _ => die
+        }
+      }
+    case Wildcard(body) =>
+      if (isQuoted) L(createASTCall("Wildcard", desugarQuote(body) :: Nil)) else R(Wildcard(desugarQuote(body)))
+    case NoCases => if (isQuoted) L(createASTCall("NoCases", Nil)) else R(NoCases)
+  }
+
+  private def toJSOperator(op: Str) = op match {
+    case "+." => "+"
+    case "-." => "-"
+    case "*." => "*"
+    case _ => op
+  }
+
+  // * Desugar `Quoted` into AST constructor invokations.
+  // * e.g., `42 will be translated into Quoted(IntLit(42)),
+  // * which will be further desugared into App(Var("IntLit"), IntLit(42)) and be traslated into `IntLit(42)` in JS
```

Comment:
Please _also_ document what a simple lambda desugars to, as it's much more interesting.

Reply:
I updated the comment.